cdap-app-templates/cdap-etl/cdap-etl-api-spark/src/main/java/co/cask/cdap/etl/api/batch/SparkMLLibContext.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.api.batch;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.data.stream.StreamBatchReadable;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.cdap.api.plugin.PluginContext;
+import co.cask.cdap.api.stream.StreamEventDecoder;
+
+import java.io.Serializable;
+import java.util.Map;
+
+/**
+ * Context passed to SparkMLLib.
+ */
+@Beta
+public interface SparkMLLibContext extends BatchContext {
+
+  /**
+   * Returns the logical start time of this Spark job. Logical start time is the time when this Spark
+   * job is supposed to start if this job is started by the scheduler. Otherwise it would be the current time when the
+   * job runs.
+   *
+   * @return Time in milliseconds since epoch time (00:00:00 January 1, 1970 UTC).
+   */
+  long getLogicalStartTime();
+
+  /**
+   * Create a Spark RDD that uses {@link Dataset} as input source
+   *
+   * @param datasetName the name of the {@link Dataset} to be read as an RDD
+   * @param kClass      the key class
+   * @param vClass      the value class
+   * @param <T>         type of RDD
+   * @return the RDD created from Dataset
+   * @throws UnsupportedOperationException if the SparkContext is not yet initialized
```

Comment:
this is thrown SparkContext methods, which we delegate to. I will look into where that happens.